### ISSUE TITLE
Integration of existing work as generator to ChipYard? What are the requirements

### ISSUE THREAD

**Issue by SihaoLiu** [ucb-bar/chipyard#1161]
### Background Work

- [X] Yes, I searched the [mailing list](https://groups.google.com/forum/#!forum/chipyard)
- [X] Yes, I searched the [documentation](https://chipyard.readthedocs.io/)

### Feature Description

We have a internal project based on https://github.com/PolyArch/dsa-framework that has been successfully integrated into ChipYard. 

It has been tested like gemmini/nvdla and works on real FPGA. I am wondering is ChipYard community considering taking external projects into the generators folder? If so, what is the requirement and how does this work?

### Motivating Example

In general, I think it will be nice for the chipyard community to be an open and diverse EDA playground for all people that are using it. A clear template for submodule-integration is very helpful.

**Comment by jerryz123** [comment]
Yes, supporting external projects is a goal we have. Unfortunately we don't have a formalized mechanism for accepting new submodules into `generators/`, but at a minimum we'd like to have some commitment to maintaining the integration layer. 

Can you show how dsa-framework was integrated into Chipyard?
